Update Test-MtCisSharedMailboxSignIn.ps1

HenrikPiecha opened the pull request:
Object generation for validating which sharedmailbox users are enabled failed / always returned all mailboxes without account enabled state in productive environments.

Changed the generation of the mailbox objects.

NZLostboy commented:
I think this one needs a few more changes, in my testing now it doesn't list any shared mailboxes if it passes. If this is by design then I think the results need to be updated to reflect that. There are two shared mailboxes in the tenant I tested against.

<img width="734" alt="Screenshot 2025-03-01 at 8 32 51 AM" src="https://github.com/user-attachments/assets/fb9b10a8-a4e5-4e83-8b2d-b5c0ad2c4c04" />

NZLostboy commented:
Also the result logic for display is broken (I did this when I originally built it), if one mailbox fails, it shows them all failed.

Example, only one of these mailboxes has sign-ins enabled:
<img width="618" alt="Screenshot 2025-03-01 at 8 44 27 AM" src="https://github.com/user-attachments/assets/b5ebabf8-69f9-4c36-abdf-221e784d36e9" />

I realise you didn't make this error, but we should fix it while we are here.

HenrikPiecha commented:
@NZLostboy Added a Skip when there are no shared mailboxes in the tenant. 
Also changed the output of the result to only print the table when it failes and changed to table to include the displayname and the upn.
I think its not mandatory to add a "failed" result to each mailbox as its only printing the failed ones.

Let me know what you think. 

BR

NZLostboy commented:
> @NZLostboy Added a Skip when there are no shared mailboxes in the tenant. Also changed the output of the result to only print the table when it failes and changed to table to include the displayname and the upn. I think its not mandatory to add a "failed" result to each mailbox as its only printing the failed ones.
> 
> Let me know what you think.
> 
> BR

Approach is fine with me, looks great thank you.

merill commented:
Thanks for raising this and the fix!
